libs/utils/include/celix_filter.h

```diff
@@ -44,63 +89,121 @@ typedef enum celix_filter_operand_enum
+ * @brief Find the filter attribute.
+ *
+ * @param[in] filter The filter.
+ * @param[in] attribute The attribute to find.
+ * @return The found attribute value or NULL if the attribute is not found. The returned string is owned by the filter,
+ *         must not be freed and is only valid as long as the filter is valid.
  */
-CELIX_UTILS_EXPORT const char* celix_filter_findAttribute(const celix_filter_t *filter, const char *attribute);
+CELIX_UTILS_EXPORT const char* celix_filter_findAttribute(const celix_filter_t* filter, const char* attribute);
 
 /**
- * Check whether the filter indicates the mandatory presence of an attribute with a specific value for the provided attribute key.
+ * @brief Check whether the filter indicates the mandatory presence of an attribute with a specific value for the
+ * provided attribute key.
  *
  * Example:
  *   using this function for attribute key "key1" on filter "(key1=value1)" yields true.
  *   using this function for attribute key "key1" on filter "(!(key1=value1))" yields false.
  *   using this function for attribute key "key1" on filter "(key1>=value1)" yields false.
  *   using this function for attribute key "key1" on filter "(|(key1=value1)(key2=value2))" yields false.
-*/
-CELIX_UTILS_EXPORT bool celix_filter_hasMandatoryEqualsValueAttribute(const celix_filter_t *filter, const char *attribute);
+ *
+ * @param[in] filter The filter.
+ * @param[in] attribute The attribute to check.
+ * @return True if the filter indicates the mandatory presence of an attribute with a specific value for the provided
+ *         attribute key, false otherwise.
+ */
+CELIX_UTILS_EXPORT bool celix_filter_hasMandatoryEqualsValueAttribute(const celix_filter_t* filter,
+                                                                      const char* attribute);
 
 /**
- * Chek whether the filter indicates the mandatory absence of an attribute, regardless of its value, for the provided attribute key.
+ * @brief Check whether the filter indicates the mandatory absence of an attribute, regardless of its value, for the
+ * provided attribute key.
  *
  * example:
  *   using this function for attribute key "key1" on the filter "(!(key1=*))" yields true.
  *   using this function for attribute key "key1" on the filter "(key1=*) yields false.
  *   using this function for attribute key "key1" on the filter "(key1=value)" yields false.
  *   using this function for attribute key "key1" on the filter "(|(!(key1=*))(key2=value2))" yields false.
+ *
+ * @param[in] filter The filter.
+ * @param[in] attribute The attribute to check.
+ * @return True if the filter indicates the mandatory absence of an attribute, regardless of its value, for the provided
+ *         attribute key, false otherwise.
  */
-CELIX_UTILS_EXPORT bool celix_filter_hasMandatoryNegatedPresenceAttribute(const celix_filter_t *filter, const char *attribute);
+CELIX_UTILS_EXPORT bool celix_filter_hasMandatoryNegatedPresenceAttribute(const celix_filter_t* filter,
```

Comment:
This also suffers from lack of well-defined semantics.

Reply:
Also updated the doxygen for this, maybe that helps